rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -48,7 +48,6 @@ Recorder::Recorder(
   // TODO(karsten1987): Use this constructor later with parameter parsing.
   // The reader, storage_options as well as record_options can be loaded via parameter.
   // That way, the recorder can be used as a simple component in a component manager.
-  throw rclcpp::exceptions::UnimplementedError();
```

Comment:
@bertaveira Nop. At least not so easy..
Will need to:
1. Create default objects for `storage_options` and `record options`
2. Create writer object and pass all these arguments  to the next constructor the same way as it's made here
https://github.com/ros2/rosbag2/blob/50e28cbab7e1cbe227b64a14b721793f6bd57216/rosbag2_py/src/rosbag2_py/_transport.cpp#L211-L221
4.  Create setter methods for `storage_options` and `record options`
5. Add coverage in unit tests

Reply:
Why was the same merge acceptable in the Galactic branch but not in Humble or Rolling? It is blocking us from going to Humble right now and this does not break anything for anyone. Simply enables classes that inherit this to have some extra access that allows it to setup and start recording internally.

1. As I can see `record_options` and `storage_options` already have default values of initialisation I think. The structures have default values so it should be initialised with those already right?
3. What do you mean by setter methods?

If I go back on the change of the throw error removal would it be acceptable? Basically would just change the writer and option structures to protected instead of private?

Reply:
@bertaveira `setters` means methods which provides API to assign `record_options` and `storage_options` after calling default constructor.
e.g. 
`set_record_options(const RecordOptions & record_options)` and 
`set_storage_options(const rosbag2_storage::StorageOptions & storage_options)`

Anyway just removing `throw rclcpp::exceptions::UnimplementedError();` is not acceptable since need to create default `writer` and pass it to the another constructor.

Also since `writer` constructor  accept `record_options` as parameter need to provide "setter" method for writer interface to be able to setup `record_options` after calling writer constructor.